Migrated demo code to new new Weaviate Python client

lalit-k opened the pull request:
Contribution to #1 

What have changed:
1. Migrated the code to new new [Weaviate Python client](https://weaviate.io/blog/collections-python-client-preview)

@iamleonie Please review the PR.

thanks!

lalit-k committed:
Migrated demo code to new new Weaviate Python client

databyjp commented on create-schema.py:
```diff
@@ -1,52 +1,45 @@
 
-print("The schema has been defined.")
\ No newline at end of file
+print("The schema has been defined.")
```
We prefer to refer to the collection definition these days rather than the `schema` in general

```suggestion
print("The 'Dog' class has been defined.")
```

databyjp commented on readme.md:
````diff
@@ -20,19 +20,24 @@ Python 3.10 or later should be installed.
+2. Install the Python packages
+   ```bash
+    pip install -r requirements.txt
+    pip install -U "git+https://github.com/weaviate/weaviate-python-client.git@v3.99.0a3#egg=weaviate-client[GRPC]"
````
I suggest moving this to `requirements.txt`. It would involve adding:

```
git+https://github.com/weaviate/weaviate-python-client.git@v3.99.0a3#egg=weaviate-client[GRPC]
```

As an additional line.

databyjp commented on readme.md:
```diff
@@ -20,19 +20,24 @@ Python 3.10 or later should be installed.
 
 ## Setup instructions
 1. Make sure you have Weaviate installed and set up. [Check out the documentation](https://weaviate.io/developers/weaviate/current/installation/index.html) for more information!
```
Step 3 actually runs Weaviate, so I suggest removing step 1.

databyjp commented on readme.md:
````diff
@@ -20,19 +20,24 @@ Python 3.10 or later should be installed.
     docker compose up
     ```
-3. Run the schema file
+4. Run the schema file
````
`Create the class definition`

databyjp commented on readme.md:
````diff
@@ -20,19 +20,24 @@ Python 3.10 or later should be installed.
     python create-schema.py
     ```
-4. Run the images to base64 file 
+5. Run the images to base64 file 
````
Convert images to base64 encodings

databyjp commented:
Hi @lalit-k - this looks good, great job. :) It ran great for me for the most part.

I have reviewed the code briefly and made a few suggestions here and there. 

Can I also suggest adding to the README a note to create a virtual environment? e.g. suggest running

```
python3 -m venv .venv
```

So that new users don't necessarily install these libraries to their system Python. 

The other point of note is that the default Flask port (5000) is now used by Macs by default! This is annoying so I propose to change it to something else like 5001 or 5002. I think the only places where this needs to be changed is the flask docker-config file, and in `application.py`. (And the README of course!)

databyjp commented:
Oh, just one more comment from a colleague - the buffer object could be passed directly to near_image and the base64 encoding is handled by the client. :)

lalit-k committed:
implemented feedback changes

lalit-k commented:
Hi @databyjp thanks very much for review comments. 

I have implemented the changes and updated the PR. 

Regarding, buffer object for near_image function, i see it's accepting the io.BufferedReader object so wrapped the buffer object into io.BufferedReader and it worked fine.
